tidy3d/components/simulation.py

```diff
@@ -2556,6 +2572,57 @@ def tfsf_with_symmetry(cls, val, values):
                 raise SetupError("TFSF sources cannot be used with symmetries.")
         return val
 
+    @staticmethod
+    def _get_fixed_angle_sources(sources: Tuple[SourceType, ...]):
+        """Get list of plane wave sources with ``FixedAngleSpec``."""
+
+        return [
+            source for source in sources if isinstance(source, PlaneWave) and source._is_fixed_angle
+        ]
+
+    @pydantic.root_validator(skip_on_failure=True)
```

Comment:
do you want to add `skip_if_missing` here? (might have the name mixed up, but validator skipping if fields (like `sources`) fail?